### PR TITLE
feat(Structured Output Parser Node): Refactor Output Parsers and Improve Error Handling

### PR DESCRIPTION
**Pull request by OlegIvaniv**
## Summary
This PR refactors our output parser implementation, extracting them into individual classes and improving error handling, particularly for structured output parsers. These changes should improve our ability to provide user-friendly error messages while maintaining the functionality of the auto-fixing parser.

- The `N8nStructuredOutputParser` now includes a `context` property to handle execution context and supports custom error mapping.
- `N8nOutputFixingParser` integrates with the structured parser to attempt fixing invalid outputs.
- `N8nItemListOutputParser` retains its functionality but is now part of the new parser structure.
- Updated `getOptionalOutputParsers` utility to work with the new parser types.

## Changes

1. Extracted output parsers into separate classes:
   - `N8nStructuredOutputParser`
   - `N8nOutputFixingParser`
   - `N8nItemListOutputParser`

2. Implemented error mapping in `N8nStructuredOutputParser` to allow for both user-friendly error messages and proper error handling in the auto-fixing parser.

3. Removed proxy wrapping from `logWrapper` and instead manually populate input/output data from each parser class.

4. Updated relevant nodes and utility functions to use the new parser classes.

5. Adjusted test files to reflect the new structure and functionality and added tests for `N8nOutputFixingParser` and `N8nItemListOutputParser` nodes.
## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/nodes-langchain/nodes/output_parser/OutputParserAutofixing/test/OutputParserAutofixing.node.test.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	it('should attempt to fix and parse invalid output', async () => {
+		const validOutput = { name: 'John', age: 30 };
+		mockFixingParser.parse.mockResolvedValue(validOutput);
+
+		const { response } = await outputParser.supplyData.call(thisArg, 0);
+		const result = await (response as N8nOutputFixingParser).parse('Invalid JSON string');
+
+		expect(result).toEqual(validOutput);
+		expect(mockFixingParser.parse).toHaveBeenCalledTimes(1);
+	});
```
Comment (burivuhster) [inline packages/@n8n/nodes-langchain/nodes/output_parser/OutputParserAutofixing/test/OutputParserAutofixing.node.test.ts:85]: Does this test really checks the attempt to fix? Probably it should emulate exception on the first parse call, and provide valid mock on a second.

File: packages/@n8n/nodes-langchain/utils/output_parsers/N8nStructuredOutputParser.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		super(zodSchema);
+		this.context = context;
+	}
+
+	lc_namespace = ['langchain', 'output_parsers', 'structured'];
+
+	async parse(
+		text: string,
+		_callbacks?: Callbacks,
+		errorMapper?: (error: Error) => Error,
```
Comment (burivuhster) [inline packages/@n8n/nodes-langchain/utils/output_parsers/N8nStructuredOutputParser.ts:29]: Are second and third parameters here actually supplied by callers?

**Review by burivuhster — CHANGES_REQUESTED** [review]
Awesome job! And kudos for writing tests.
I have only one comment about auto-fixing parser test case, could you please check?

**Review by burivuhster — DISMISSED** [review]

**Review by burivuhster — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.65.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.65.0)

**Pull request merged**